io/net/virt-net/lpm.py

```diff
@@ -0,0 +1,329 @@
+        self.ports = self.params.get("sriov_ports",
+                                     default='').split(',')
+        self.remote_ports = self.params.get("remote_sriov_ports",
+                                            default='').split(',')
+
+        self.adapter_id = []
+        for adapter in self.adapters:
+            self.adapter_id.append(self.get_adapter_id(self.server, adapter))
+        self.remote_adapter_id = []
+        for adapter in self.remote_adapters:
+            self.remote_adapter_id.append(
+                self.get_adapter_id(self.remote_server, adapter))
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=3000):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc: %s", command)
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        return output
+
+    def rsct_service_start(self):
+        '''
+        Running rsct services which is necessary for Network
+        virtualization tests
+        '''
+        try:
+            for svc in ["rsct", "rsct_rm"]:
+                process.run('startsrc -g %s' % svc, shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Starting service %s failed", svc)
+
+        output = process.system_output("lssrc -a", ignore_status=True,
+                                       shell=True, sudo=True)
+        if "inoperative" in output:
```

Comment:
it this check really required as we are capturing any failure to start service and failing test case

Reply:
We are doing a group start of services. So, either the command can fail (handled via exception), or via here.
I will either fail both or cancel both, to maintain consistency.